File: Rebex Components R6.11 - Samples/net-3.5/Mail/MimeExplorer_VB/ValidationForm.vb
'  
'   Rebex Sample Code License
' 
'   Copyright 2023, REBEX CR s.r.o.
'   All rights reserved.
'   https://www.rebex.net/
' 
'   Permission to use, copy, modify, and/or distribute this software for any
'   purpose with or without fee is hereby granted.
' 
'   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
'   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
'   OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
'   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
'   HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
'   WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
'   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
'   OTHER DEALINGS IN THE SOFTWARE.
' 

Imports Rebex.Security.Certificates
Imports Rebex.Security.Cryptography.Pkcs

Public Class ValidationForm
    Inherits System.Windows.Forms.Form

    Public Sub New(ByVal result As SignatureValidationResult)
        MyBase.New()
        InitializeComponent()

        If result Is Nothing Then
            Return
        End If

        Dim certStatus As ValidationStatus = result.CertificateValidationStatus
        Dim vs As ValidationStatus
        For Each vs In [Enum].GetValues(GetType(ValidationStatus))
            If (vs And certStatus) <> 0 Then
                textCertProblem.Text += ValidationHelper.GetCertificateValidationStatusDescription(vs) + vbCrLf
            End If
        Next

        Dim signStatus As SignatureValidationStatus = result.Status
        Dim ss As SignatureValidationStatus
        For Each ss In [Enum].GetValues(GetType(SignatureValidationStatus))
            If (ss And signStatus) <> 0 Then
                textSignProblem.Text += ValidationHelper.GetSignatureValidationStatusDescription(ss) + vbCrLf
            End If
        Next
    End Sub

#Region " Windows Form Designer generated code "

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    Friend WithEvents textSignProblem As System.Windows.Forms.TextBox
    Friend WithEvents label1 As System.Windows.Forms.Label
    Friend WithEvents panel2 As System.Windows.Forms.Panel
    Friend WithEvents textCertProblem As System.Windows.Forms.TextBox
    Friend WithEvents label6 As System.Windows.Forms.Label
    Friend WithEvents btnClose As System.Windows.Forms.Button
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Me.textSignProblem = New System.Windows.Forms.TextBox
        Me.label1 = New System.Windows.Forms.Label
        Me.panel2 = New System.Windows.Forms.Panel
        Me.textCertProblem = New System.Windows.Forms.TextBox
        Me.label6 = New System.Windows.Forms.Label
        Me.btnClose = New System.Windows.Forms.Button
        Me.SuspendLayout()
        '
        'textSignProblem
        '
        Me.textSignProblem.BackColor = System.Drawing.SystemColors.ControlLightLight
        Me.textSignProblem.Location = New System.Drawing.Point(8, 160)
        Me.textSignProblem.Multiline = True
        Me.textSignProblem.Name = "textSignProblem"
        Me.textSignProblem.ReadOnly = True
        Me.textSignProblem.ScrollBars = System.Windows.Forms.ScrollBars.Vertical
        Me.textSignProblem.Size = New System.Drawing.Size(384, 88)
        Me.textSignProblem.TabIndex = 14
        Me.textSignProblem.Text = ""
        '
        'label1
        '
        Me.label1.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(238, Byte))
        Me.label1.Location = New System.Drawing.Point(8, 8)
        Me.label1.Name = "label1"
        Me.label1.Size = New System.Drawing.Size(384, 24)
        Me.label1.TabIndex = 13
        Me.label1.Text = "Signature validation problems:"
        '
        'panel2
        '
        Me.panel2.BackColor = System.Drawing.SystemColors.ControlDarkDark
        Me.panel2.Location = New System.Drawing.Point(8, 128)
        Me.panel2.Name = "panel2"
        Me.panel2.Size = New System.Drawing.Size(380, 1)
        Me.panel2.TabIndex = 5
        '
        'textCertProblem
        '
        Me.textCertProblem.BackColor = System.Drawing.SystemColors.ControlLightLight
        Me.textCertProblem.Location = New System.Drawing.Point(8, 32)
        Me.textCertProblem.Multiline = True
        Me.textCertProblem.Name = "textCertProblem"
        Me.textCertProblem.ReadOnly = True
        Me.textCertProblem.ScrollBars = System.Windows.Forms.ScrollBars.Vertical
        Me.textCertProblem.Size = New System.Drawing.Size(384, 88)
        Me.textCertProblem.TabIndex = 12
        Me.textCertProblem.Text = ""
        '
        'label6
        '
        Me.label6.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(238, Byte))
        Me.label6.Location = New System.Drawing.Point(8, 136)
        Me.label6.Name = "label6"
        Me.label6.Size = New System.Drawing.Size(392, 24)
        Me.label6.TabIndex = 11
        Me.label6.Text = "Certificate validation problems:"
        '
        'btnClose
        '
        Me.btnClose.FlatStyle = System.Windows.Forms.FlatStyle.System
        Me.btnClose.Location = New System.Drawing.Point(312, 256)
        Me.btnClose.Name = "btnClose"
        Me.btnClose.TabIndex = 6
        Me.btnClose.Text = "Close"
        '
        'ValidationForm
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(394, 288)
        Me.Controls.Add(Me.btnClose)
        Me.Controls.Add(Me.label6)
        Me.Controls.Add(Me.panel2)
        Me.Controls.Add(Me.textSignProblem)
        Me.Controls.Add(Me.label1)
        Me.Controls.Add(Me.textCertProblem)
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        Me.MinimizeBox = False
        Me.MaximizeBox = False
        Me.ShowInTaskbar = False
        Me.MinimizeBox = False
        Me.MaximizeBox = False
        Me.MaximumSize = New System.Drawing.Size(424, 332)
        Me.MinimumSize = New System.Drawing.Size(424, 332)
        Me.Name = "ValidationForm"
        Me.Text = "Signature Is Not Valid"
        Me.ResumeLayout(False)

    End Sub

#End Region

    Private Sub btnClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

End Class

File: Rebex Components R6.11 - Samples/net-3.5/Mail/MimeExplorer_CS/ValidationForm.cs
//  
//   Rebex Sample Code License
// 
//   Copyright 2023, REBEX CR s.r.o.
//   All rights reserved.
//   https://www.rebex.net/
// 
//   Permission to use, copy, modify, and/or distribute this software for any
//   purpose with or without fee is hereby granted.
// 
//   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
//   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
//   OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
//   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
//   HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
//   WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
//   OTHER DEALINGS IN THE SOFTWARE.
// 

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using Rebex.Security.Certificates;
using Rebex.Security.Cryptography.Pkcs;
using Rebex.Samples;

namespace Rebex.Samples
{
    /// <summary>
    /// Summary description for ValidationForm.
    /// </summary>
    public class ValidationForm : System.Windows.Forms.Form
    {
        private System.Windows.Forms.Button btnClose;
        private System.Windows.Forms.TextBox textSignProblem;
        private System.Windows.Forms.Label label1;
        private System.Windows.Forms.Panel panel2;
        private System.Windows.Forms.TextBox textCertProblem;
        private System.Windows.Forms.Label label6;

        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components = null;

        public ValidationForm(SignatureValidationResult result)
        {
            InitializeComponent();

            if (result == null)
                return;

            ValidationStatus certStatus = result.CertificateValidationStatus;
            foreach (ValidationStatus status in Enum.GetValues(typeof(ValidationStatus)))
            {
                if ((status & certStatus) != 0)
                    textCertProblem.Text += ValidationHelper.GetCertificateValidationStatusDescription(status) + "\r\n";
            }

            SignatureValidationStatus signStatus = result.Status;
            foreach (SignatureValidationStatus status in Enum.GetValues(typeof(SignatureValidationStatus)))
            {
                if ((status & signStatus) != 0)
                    textSignProblem.Text += ValidationHelper.GetSignatureValidationStatusDescription(status) + "\r\n";
            }
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose( bool disposing )
        {
            if( disposing )
            {
                if(components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose( disposing );
        }

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.btnClose = new System.Windows.Forms.Button();
            this.textSignProblem = new System.Windows.Forms.TextBox();
            this.label1 = new System.Windows.Forms.Label();
            this.panel2 = new System.Windows.Forms.Panel();
            this.textCertProblem = new System.Windows.Forms.TextBox();
            this.label6 = new System.Windows.Forms.Label();
            this.SuspendLayout();
            // 
            // btnClose
            // 
            this.btnClose.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.btnClose.Location = new System.Drawing.Point(312, 256);
            this.btnClose.Name = "btnClose";
            this.btnClose.TabIndex = 0;
            this.btnClose.Text = "Close";
            this.btnClose.Click += new System.EventHandler(this.btnClose_Click);
            // 
            // textSignProblem
            // 
            this.textSignProblem.BackColor = System.Drawing.SystemColors.ControlLightLight;
            this.textSignProblem.Location = new System.Drawing.Point(8, 160);
            this.textSignProblem.Multiline = true;
            this.textSignProblem.Name = "textSignProblem";
            this.textSignProblem.ReadOnly = true;
            this.textSignProblem.ScrollBars = System.Windows.Forms.ScrollBars.Vertical;
            this.textSignProblem.Size = new System.Drawing.Size(384, 88);
            this.textSignProblem.TabIndex = 14;
            this.textSignProblem.Text = "";
            // 
            // label1
            // 
            this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(238)));
            this.label1.Location = new System.Drawing.Point(8, 8);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(384, 24);
            this.label1.TabIndex = 13;
            this.label1.Text = "Signature validation problems:";
            // 
            // panel2
            // 
            this.panel2.BackColor = System.Drawing.SystemColors.ControlDarkDark;
            this.panel2.Location = new System.Drawing.Point(8, 128);
            this.panel2.Name = "panel2";
            this.panel2.Size = new System.Drawing.Size(380, 1);
            this.panel2.TabIndex = 5;
            // 
            // textCertProblem
            // 
            this.textCertProblem.BackColor = System.Drawing.SystemColors.ControlLightLight;
            this.textCertProblem.Location = new System.Drawing.Point(8, 32);
            this.textCertProblem.Multiline = true;
            this.textCertProblem.Name = "textCertProblem";
            this.textCertProblem.ReadOnly = true;
            this.textCertProblem.ScrollBars = System.Windows.Forms.ScrollBars.Vertical;
            this.textCertProblem.Size = new System.Drawing.Size(384, 88);
            this.textCertProblem.TabIndex = 12;
            this.textCertProblem.Text = "";
            // 
            // label6
            // 
            this.label6.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(238)));
            this.label6.Location = new System.Drawing.Point(8, 136);
            this.label6.Name = "label6";
            this.label6.Size = new System.Drawing.Size(384, 24);
            this.label6.TabIndex = 11;
            this.label6.Text = "Certificate validation problems:";
            // 
            // ValidationForm
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(394, 288);
            this.Controls.Add(this.btnClose);
            this.Controls.Add(this.panel2);
            this.Controls.Add(this.label6);
            this.Controls.Add(this.textCertProblem);
            this.Controls.Add(this.label1);
            this.Controls.Add(this.textSignProblem);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.MinimizeBox = true;
            this.MaximizeBox = false;
            this.ShowInTaskbar = false;
            this.MaximumSize = new System.Drawing.Size(424, 332);
            this.MinimumSize = new System.Drawing.Size(424, 332);
            this.Name = "ValidationForm";
            this.Text = "Signature Is Not Valid";
            this.ResumeLayout(false);

        }
        #endregion

        private void btnClose_Click(object sender, System.EventArgs e)
        {
            this.Close();
        }
    }
}
